### DISCUSSION TITLE
Downloadhelper 9.1.0.42 stuck in queue

### DISCUSSION THREAD

**Discussion by harapra** [aclap-dev/video-downloadhelper#1217] [Help and Q&A]
### Your question

I recently upgraded Ubuntu 22.04 to the latest LTS 24.04 and with it came the change in the version of Video Downloadhelper to V9.1xxx. Until then everything worked as expected.
I have tried several way to get the Downloadhelper to work, helas in vain.
Things I tried:
Extension in extensionmanager (allowin to run in private windows)
curl ................. 
and in stalling the compagnion app (which keeps being verified)

if I try to download a movie, I tried a Lego movie, the blue download button changes to red and keep circling. Nothing is downloaded to the dwhelper directory.

What have I missed or done wrong.

Firefox is the deb version 130.

### Addon & CoApp version and details

Version 9.1.0.42 (stable), target mozilla, locale en-US.

Verifying Companion App... (circle turning)

**Comment by Mattmjs** [comment]
It look like the extension is not seeing the companion app if it's staying on Verifying. I believe you also have to accept a prompt
in Firefox on Ubuntu. Did you get this prompt ?

**Reply by harapra** [reply]
Thank you for answering, Mattmjs, yes I got the prompt when installing the extension. I obviously confirmed and also allowed to run in private windows, which I use all the time.

**Reply by Mattmjs** [reply]
Sounds like if it's stuck on verifying and doesn't see the companion app Ubuntu must have broke something when you 
went to version 24.04.

**Reply by harapra** [reply]
Hello Mattmijs,
this is highly unlikely, since I re-installed everything, concerning de downloadhelper, even several times after doing the release upgrade. I even changed from the snap version of Firefox to the deb version. I hope some other explanation and/ or solution will come up.

**Reply by Mattmjs** [reply]
I think someone else needs to try Ubuntu 24.04 or maybe the dev @paulrouget  will need to investigate it.

**Comment by Mattmjs** [comment]
I ran 24.04.1 of Ubuntu as a live disk in hyper v , added extension to Firefox then installed the companion app using
the curl command. Accepted the prompt when it appeared. The extension is able to see the companion app.
A download completed, so it all seems to be working. Did you refer to the troubleshooting page, there is a command to
re-register the app.

**Reply by harapra** [reply]
Thank you again, Mattmjs. I went even further, uninstalled the lot including Firefox end re-installed then the deb version, went to the VDH page installed fro there to finally install the coapp which I re-registered after it did not do a lot in the first try.

https://github.com/user-attachments/assets/c1560f18-7fdd-4052-b581-f90a07c2956b

This brief movie will show you what I see. Not a lot. I don't get it . I ran VDH for many years already over many Ubuntu version (my first was in 2007).

**Reply by Mattmjs** [reply]
I don't see anything from the mp4 you added here, try showing some screenshots of what you've done so far.
Also from the Wiki, run this command in the terminal and post the information here.  ~/.local/share/vdhcoapp/vdhcoapp --info

**Reply by harapra** [reply]
Hi Mattmjs, this is the result of install and the re-registering. Everything looks just fine:

The video, you unfortunately cannot see, is just showing the VDH depicting the red icon on the right side of the download line during download running in circles bur unfortunately nothing gets downloaded. Not very exciting. That's why it I kept it so short.

Sudo apt install firefox -- installed the browser, as you might have guessed ;-)
![image](https://github.com/user-attachments/assets/36fe983e-6d93-4e56-9bea-18a4d5f0aac0)
![image](https://github.com/user-attachments/assets/9bf33d81-5506-4c55-81ba-33669b82cb7f)

---------------------------------
harald@harald-Ryzen7:~$ ~/.local/share/vdhcoapp/vdhcoapp install
Installing…
Writing /home/harald/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
Writing /home/harald/.var/app/org.mozilla.firefox/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
VdhCoApp : VdhCoApp is ready to be used
---------------------------------------------------

**Reply by harapra** [reply]
Hi, I just had a look into the legacy interface. It says, "queued" in the list of resolutions to download and orphaned in the pop-up that has the the settings wheel on the right. Maybe this gives a hint. I am pretty sure the presence of a video is not recognised. I just do not understand why.

**Comment by Mattmjs** [comment]
I'll show you what I see in Ubuntu and then you can show  how it looks in yours.

Terminal installation, you've shown this and looks the same here.
![Screenshot from 2024-09-11 12-03-23](https://github.com/user-attachments/assets/113ca35f-5a4d-42a0-93d9-0ba88d40564a)

VDH settings showing the extension can see the companion app, you haven't shown this.
![Screenshot from 2024-09-11 11-57-38](https://github.com/user-attachments/assets/63fc8b6b-d0d7-475d-9f0d-b561917680ee)

Two completed downloads, which I launched around the same time. Nothing got queued.
What's unusual is that I used the stable extension and each file has their own name.
On windows the stable extension has an issue with the file names seen in another discussion.
![Screenshot from 2024-09-11 12-18-59](https://github.com/user-attachments/assets/9c4b28e6-defc-4ea6-ab01-21991c1d9179)

**Reply by harapra** [reply]
Hi, here's what I get:
![image](https://github.com/user-attachments/assets/b51c615f-dffe-45a3-b545-488856eedf03)
and:
![image](https://github.com/user-attachments/assets/fe112466-872f-449e-ba8f-646290d1e2cf)
Unfortunately I never get the download, well the co-app is not recognised. I guess I'll have to remove everything from Firefox and start over. This is somewhat frustrating.

**Reply by harapra** [reply]
By-the-way: Is your installation a Flatpack or a deb?

**Reply by Mattmjs** [reply]
You mean if installed companion app is flatpack or deb, I don't know but it works for me. I tried to install the companion app
manually by downloading the .deb 64bits installer but I was stumped what to do next. I have done it manually in linux mint.
You could try it manually but first refer to the wiki for Ubuntu and follow what it says. Then come back with the results.
https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems

**Reply by harapra** [reply]
Eureca, after having removed every trace of firefox, mozilla and all king of extensions I re-installed the lot. All install files are from as basic as can like only deb versions (I did not re-compile things). Guess what. It works now.
Unfortunately this still leaves me with the question: why? I was just too frustrated to keep searching for the reason. Thank you so much for your encouraging words, your patience and help to find a solution. Everything is deb based and has to be because otherwise I would not be able to connect to government services e.g. the ministry of finance. A lot of the communication over here is digitised, so we have to.
thanks again.
